api/type.rb

```diff
@@ -494,6 +494,7 @@ class NameValues < Composite
 
       def validate
         super
+        default_value_property :key_type, Api::Type::String.to_s
```

Comment:
Do you need that and `check_property`?

Reply:
Yeah. If you override the value (for whatever reason), we need to check that the value is still of type ::String